core-bundle/contao/templates/modules/mod_login.html5

```diff
@@ -15,6 +15,10 @@
       <input type="hidden" name="REQUEST_TOKEN" value="<?= $this->requestToken ?>">
       <input type="hidden" name="_target_path" value="<?= $this->targetPath ?>">
       <input type="hidden" name="_always_use_target_path" value="<?= $this->forceTargetPath ?>">
+      <?php if ($this->remembered): ?>
+        <input type="hidden" name="username" value="<?= $this->value ?>">
```

Comment:
This looks like something that can be manipulated. Since there is a user already, can we not omit this entirely?

Reply:
we cannot, because the username must be submitted for the firewall authenticator to accept it. However, I don't think manipulation is a problem here, because if you enter a different username, you would need to know the password of that user, and you would simply be logged in as someone else then.

Reply:
> must be submitted for the firewall authenticator to accept it

Since we have our own `ContaoLoginAuthenticator`, we could get the username from the token in the `getCredentials()` method if it has not been submitted, right?